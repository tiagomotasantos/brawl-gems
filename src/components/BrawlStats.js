import React from 'react';

const BrawlStats = ({ label, icon, value}) => (
    <div className="brawl-stats">
        <img alt={label} src={icon} className="brawl-stats__icon" />
        <span className="brawl-stats__label">{ label }</span>
        <span className="brawl-stats__value">{ value }</span>
    </div>
);

export default BrawlStats;