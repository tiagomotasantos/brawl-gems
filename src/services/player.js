
import axios from './axios';

export const getPlayer = (tag) => {
    return axios.get(`player?tag=${tag}`);
}