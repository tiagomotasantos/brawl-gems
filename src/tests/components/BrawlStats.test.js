import React from 'react';
import { shallow } from 'enzyme';
import BrawlStats from '../../components/BrawlStats';
import player from '../fixtures/player';

test('should render BrawlStats', () => {
    const wrapper = shallow(<BrawlStats 
        label="Trophies"
        icon="/images/trophy.png" 
        value={player.trophies}
    />);
    expect(wrapper).toMatchSnapshot();
});