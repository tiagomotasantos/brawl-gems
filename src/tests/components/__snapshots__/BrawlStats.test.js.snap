// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BrawlStats 1`] = `
<div
  className="brawl-stats"
>
  <img
    alt="Trophies"
    className="brawl-stats__icon"
    src="/images/trophy.png"
  />
  <span
    className="brawl-stats__label"
  >
    Trophies
  </span>
  <span
    className="brawl-stats__value"
  >
    6262
  </span>
</div>
`;
